src/kfactory/kcell.py

```diff
@@ -3000,7 +3000,11 @@ def __getattr__(self, name):  # type: ignore[no-untyped-def]
     def name(self) -> str | None:
         """Name of instance in GDS."""
         prop = self.property(PROPID.NAME)
-        return str(prop) if prop is not None else None
+        return (
+            str(prop)
+            if prop is not None
+            else f"{self.kcl[self.cell_index].name}_{self.x}_{self.y}"
+        )
```

Comment:
Why return a name if it's not a named instance? This is a bit weird, because it doesn't necessarily reflect what there is and also it's not necessarily unique. I can create you two connected waveguides that will return the same instance name, but are properly connected waveguides.

Reply:
otherwise if a component accepts an instance, then `clean_value(instance)` will return None and it will fail when trying to replace the invalid characters `,|.|?`
I think giving them default names is also what we did in gdsfactory, it makes it nicer for circuit simulations